### ISSUE TITLE
csv to column-oriented flat file?

### ISSUE THREAD

**Issue by dagoldman** [wireservice/csvkit#1152]
I saw a previous post maybe on this topic, but needed more information.

Suppose a csv file is as follows:
```
"asthma",37
"bacterial sepsis",22
```
I need to convert to column-oriented flat file. To be clear what I need, below is an example, with in this case the first field in columns 1-20 and the second field in columns 21-23:
```
asthma               37
bacterial sepsis     22
```

First question: Does csvkit convert csv to a column-oriented flat file? Not a criticism. csvkit seems very good. Just verifying that it does not output a column-oriented flat file.

Second question: Assuming csvkit does not convert csv to column-oriented flat file, can you please suggest an alternative open-source software that does?

Thanks,
Daniel

**Comment by dagoldman** [comment]
I see that csvlook makes a kind of fixed-width table. Besides my previous questions, can someone explain what "Markdown-compatible" means? My guess is [https://en.wikipedia.org/wiki/Markdown] language. Also, does cvslook allow specifying the columns for each field in the output table? My guess is that it does not allow. I am also looking at cvstool, which I believe I saw mentioned on the cvskit website. Thanks

**Comment by dagoldman** [comment]
I think neither csvkit nor csvtool make a column-oriented flat file. But maybe that is not a problem. They both cut a column from a csv file. The cut column can then be manipulated, for example with sed, to make the column a set length. Then the column can be pasted with other columns to make the desired column-oriented flat file. The key step is correctly cutting the column, and both programs seem to do that OK, as well as doing many other operations. 

csvtool is a lot faster for cutting columns, in case you are processing a lot of records. I made a large test file with the following script:

```
rm -f /tmp/temp.x
n=1
while [ $n -le 10000000 ]; do
  echo "100,200,300,400,500" >> /tmp/temp.x
  let "n++"
done
```

Then I ran the following a few times:

```
time csvtool col 3 /tmp/temp.x > /dev/null
time csvcut -c 3 /tmp/temp.x > /dev/null
```

Here are the typical times I got:

```
csvtool 
real    0m4.084s
user    0m4.064s
sys     0m0.020s

csvcut
real    0m15.250s
user    0m15.186s
sys     0m0.064s
```
There is another open-source program called ffe (flat file extractor) that also looks good. ffe requires the use of configuration files and has many options. I believe it may produce an output column-oriented flat file in one operation, but I am not sure yet, because it is a little difficult figuring out all the options.

Finally, there is libcsv which has about 20 functions for parsing and writing csv data. It does not seem trivial to learn.

Daniel
